src/sql/workbench/performance/browser/ADSPerformance.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface ADSPerformanceEntry {
+	readonly name: string;
+	readonly timestamp: number;
+}
+
+export class ADSPerformance {
+	static browserPerformance: Performance = window.performance;
```

Comment:
Are extensions going to have access to this at all? If not we may want to consider adding an extension point to enable doing so - being able to use this for extensions would be great.

Reply:
That's a great idea - since this is static we'd have to wrap around this and prepend the extension name to the tags being pushed in.